compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -480,7 +480,28 @@ createProjectPackageDb ::
 createProjectPackageDb (AllowDifferentSdkVersions allowDiffSdkVersions) lfVersion fps = do
     let dbPath = projectPackageDatabase </> lfVersionString lfVersion
     createDirectoryIfMissing True $ dbPath </> "package.conf.d"
-    let fps0 = filter (`notElem` basePackages) fps
+    -- Expand SDK package dependencies using the SDK root path.
+    -- E.g. `daml-trigger` --> `$DAML_SDK/daml-libs/daml-trigger.dar`
+    -- Or, fail if not run from DAML assistant.
+    mbSdkPath <- handleIO (\_ -> pure Nothing) $ Just <$> getSdkPath
+    let isSdkPackage fp = takeExtension fp /= ".dar"
+        handleSdkPackages :: [FilePath] -> IO [FilePath]
+        handleSdkPackages = case mbSdkPath of
```

Comment:
Might be easier to read if you do the case split inside of `expand`.

Reply:
Agreed that's clearer, I've changed the code.

I figured lifting the case match outside was probably more efficient. But, that's likely premature optimization and not really necessary.